runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -3329,13 +3329,25 @@ static void exit_any(int status) {
 }
 
 
+//
+// FIXME: this is a workaround for hangs when the provider is EFA:
+//        we skip fi_close and don't properly shutdown libfabric
+//
+static inline
+int chpl_fi_close_wrapper(chpl_bool using_efa, struct fid* fid) {
```

Comment:
I would call `providerInUse` from within `chpl_fi_close_wrapper` as it avoids the `using_efa` parameter. `using_efa` is probably less overhead, but we're shutting down so it's not really on the critical path.